azure-kusto-data/azure/kusto/data/response.py

```diff
@@ -90,6 +74,30 @@ def __len__(self) -> int:
         return self.tables_count
 
 
+class KustoResponseDataSet(BaseKustoResponseDataSet, metaclass=ABCMeta):
+    """
+    `KustoResponseDataSet` Represents the parsed data set carried by the response to a Kusto request.
+    `KustoResponseDataSet` provides convenient methods to work with the returned result.
+    The result table(s) are accessable via the @primary_results property.
+    @primary_results returns a collection of `KustoResultTable`.
+        It can contain more than one table when [`fork`](https://docs.microsoft.com/en-us/azure/kusto/query/forkoperator) is used.
+    """
+
+    def __init__(self, json_response):
+        self.tables = [KustoResultTable(t) for t in json_response]
+        self.tables_count = len(self.tables)
+        self.tables_names = [t.table_name for t in self.tables]
+
+    @property
+    def primary_results(self) -> List[KustoResultTable]:
+        """Returns primary results. If there is more than one returns a list."""
+        if self.tables_count == 1:
+            return self.tables
+        primary = list(filter(lambda x: x.table_kind == WellKnownDataSet.PrimaryResult, self.tables))
```

Comment:
why not a list comprehension instead of the list/filter?

Reply:
![image](https://user-images.githubusercontent.com/6424271/135976972-56cfb185-a497-4d6e-aca4-a7081226500a.png) ;)